risc0/zkvm/prove/method_id.cpp

```diff
@@ -23,20 +24,38 @@
 
 namespace risc0 {
 
-MethodID makeMethodID(const std::string& elfFile) {
-  // Load the ELF file into an image
+MethodId makeMethodId(const MethodDigest& digest) {
+  MethodId id;
+  std::memcpy(&id, &digest, sizeof(MethodId));
+  return id;
+}
+
+MethodId makeMethodId(const uint8_t* bytes, const size_t len) {
+  if (len != sizeof(MethodId)) {
+    throw std::length_error("Got buffer of invalid size!");
+  }
+  MethodId id;
+  std::memcpy(&id, bytes, sizeof(MethodId));
+  return id;
+}
+
+MethodId makeMethodId(const std::string& elfPath) {
+  return makeMethodId(makeMethodDigest(elfPath));
+}
+
+MethodDigest makeMethodDigest(const std::string& elfPath) {
   std::map<uint32_t, uint32_t> image;
-  uint32_t startAddr = loadElf(elfFile, kMemSize, image);
+  uint32_t startAddr = loadElf(elfPath, kMemSize, image);
 
   // Start with an empty return value
-  MethodID ret;
+  MethodDigest digest;
 
   // Make the digest for each level
-  for (size_t i = 0; i < kCodeDigestCount; i++) {
+  for (size_t i = 0; i < numMethodDigests; i++) {
```

Comment:
In C++, use `kSomething` for constants (ala Google coding style).

I think `kCodeDigestCount` is a reasonable name here.

Reply:
Ah. I don't even think that "rename" was intentional. Will change back.